test/ecto/query/planner_test.exs

```diff
@@ -251,6 +252,35 @@ defmodule Ecto.Query.PlannerTest do
     assert params == []
   end
 
+  test "plan: casts atom values" do
+    {_query, params, _key} = plan(Post |> where([p], p.status == :draft))
+    assert params == ["draft"]
```

Comment:
So this example in particular should not have any parameter, instead, we will convert :draft into something directly in the query.